### PR TITLE
Implement Upcoming Launches

### LINKED ISSUE #47: Implement Remaining Launches Endpoints
The following Launches endpoints are still to be implemented:

- [x] ~[One Launch](https://documenter.getpostman.com/view/2025350/RWaEzAiG#cfcc49e7-5fe4-4dd3-9701-7c5caf7af9fb)~ - #48
- [x] ~[Past Launches](https://documenter.getpostman.com/view/2025350/RWaEzAiG#fce450d6-e064-499a-b88d-34cc22991bcc)~ - #52
- [ ] [Upcoming Launches](https://documenter.getpostman.com/view/2025350/RWaEzAiG#e001c501-9c09-4703-9e29-f91fbbf8db7c)

Reference: https://documenter.getpostman.com/view/2025350/RWaEzAiG#bc65ba60-decf-4289-bb04-4ca9df01b9c1

### PR DESCRIPTION
**Pull request by mtking2**
* Implements the [Upcoming Launches](https://documenter.getpostman.com/view/2025350/RWaEzAiG#e001c501-9c09-4703-9e29-f91fbbf8db7c) ability of the Launches endpoint that allows you to request all future scheduled launches.
* Finishes and closes #47

### CODE REVIEW

**Comment by rodolfobandeira** [comment]
@mtking2 Sorry but I ended merging PR #57 first. Can you please merge the latest master and fix the conflict? 
Thanks again @mtking2

File: README.md [hunk 1]
```diff
@@ -314,6 +315,22 @@ past_launches.last.rocket.first_stage.cores.first.land_success # true
 past_launches.last.launch_success # true
 ```
 
+This code snippet shows how to get information on upcoming launches:
```
Comment (invacuo) [inline README.md]: Do you mind following the pattern that the other examples are using? Just to keep it consistent?

File: spec/spacex/launches_spec.rb [hunk 2]
```diff
@@ -393,6 +393,27 @@
...
+      SPACEX::Launches.upcoming
+    end
+
+    it 'returns and array of upcoming launch hashes' do
+      expect(subject).to be_an Array
+      expect(subject.first).to be_a Hash
+      expect(subject.all?(&:upcoming)).to be true
+    end
+
+    it 'returns launches scheduled in the future' do
```
Comment (invacuo) [inline spec/spacex/launches_spec.rb]: I like these tests. Can you also add one that tests to make sure we get all the expected fields for each launch? Specs for other end points are doing it and it helps us if ever the actual endpoint changes the data being returned (when we rerecord the cassette of course)

File: README.md (hunk 1 above)
Comment (mtking2) [reply README.md]: Sure thing. These should be consistent with the rest of the README now.

File: spec/spacex/launches_spec.rb (hunk 2 above)
Comment (mtking2) [reply spec/spacex/launches_spec.rb]: Alright added some tests to check all of the fields. Let me know if that's good.

File: .rubocop_todo.yml [hunk 3]
```diff
@@ -12,12 +12,12 @@ Lint/ParenthesesAsGroupedExpression:
     - 'spec/spacex/dragon_capsules_spec.rb'
     - 'spec/spacex/ships_spec.rb'
 
-# Offense count: 30
+# Offense count: 36
```
Comment (invacuo) [inline .rubocop_todo.yml]: Does this file need to be committed?

Comment (mtking2) [reply .rubocop_todo.yml]: I believe so in order to get the rubocop check in travis to pass. We might want to double check with @rodolfobandeira though.

Comment (rodolfobandeira) [reply .rubocop_todo.yml]: Yeah, this is usually added automatically after running `bundle exec rake rubocop gen-auto-config`.

It is usually being used to override rules like large testing blocks. So as long it doesn't override something terrible I'm OK with that. 😄

**Pull request merged**